### PR TITLE
Add display for non-passthrough customs tax

### PR DESCRIPTION
**Pull request by LordZardeck**
This depends on #160 and #161

### CODE REVIEW

File: Framework/Interaction/Tax/Get.php [hunk 1]
```diff
@@ -181,22 +173,50 @@ public function getTaxDetailsForQuote(
...
                 $taxDetails = $baseTaxDetails;
             }
 
+            $avaTaxMessages = [];
+
+            if($getTaxResult->getMessages() !== null) {
+                $landedCostMessages = array_filter(
+                    $getTaxResult->getMessages(),
+                    function ($message) {
+                        return \in_array($message->getRefersTo(), ['Customs', 'LandedCost']);
```
Comment (erikhansen) [inline Framework/Interaction/Tax/Get.php]: Move this array into a constant

File: Helper/CustomsConfig.php [hunk 2]
```diff
@@ -33,6 +33,17 @@ class CustomsConfig extends AbstractHelper
 
     const PRODUCT_ATTR_CROSS_BORDER_TYPE = 'avatax_cross_border_type';
 
+    /**
```
Comment (erikhansen) [inline Helper/CustomsConfig.php]: Switch to using this DocBlock syntax:

```
    /**#@+
     * Module config settings
     */
     …
     /**#@-*/
```

See this for reference: https://github.com/classyllama/ClassyLlama_AvaTax/blob/1c1e7122fc05b0642b683ada17b1125ec7218f41/Helper/Config.php#L36..L146

File: Model/Plugin/CartTotalRepository.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
```
Comment (erikhansen) [inline Model/Plugin/CartTotalRepository.php]: Fix copyright

File: Model/Plugin/CartTotalRepository.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * @param TotalRepository $subject
+     * @param TotalsInterface $totals
+     * @param int             $cartId
+     *
+     * @return TotalsInterface
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function afterGet(TotalRepository $subject, TotalsInterface $totals, $cartId)
```
Comment (erikhansen) [inline Model/Plugin/CartTotalRepository.php]: Add a comment somewhere, explaining the purpose for this plugin

File: Setup/UpgradeData.php [hunk 5]
```diff
@@ -182,6 +205,50 @@ public function upgrade(ModuleDataSetupInterface $setup, ModuleContextInterface
...
+
+            /** @var AttributeSet $attributeSet */
+            $attributeSet = $this->attributeSetFactory->create();
+            $attributeGroupId = $attributeSet->getDefaultGroupId($attributeSetId);
+            $customerSetup->addAttribute(
+                Customer::ENTITY,
+                CustomsConfig::CUSTOMER_IMPORTER_OF_RECORD_ATTRIBUTE,
+                [
+                    'type' => 'text',
+                    'label' => 'Override Avatax "Is Seller Importer of Record” setting',
```
Comment (erikhansen) [inline Setup/UpgradeData.php]: Change `”` smart quote to `"`

File: Ui/Component/Listing/Columns/CrossBorderClassActions.php [hunk 6]
```diff
@@ -57,6 +67,20 @@ public function prepareDataSource(array $dataSource)
...
+                /**
+                 * overriding cross border type id int with the label
+                 */
+                if(isset($item[CrossBorderClassInterface::CROSS_BORDER_TYPE])) {
+                    try {
+                        $item[CrossBorderClassInterface::CROSS_BORDER_TYPE] = $this->fetchCrossBorderTypeValue(
+                            $item[CrossBorderClassInterface::CROSS_BORDER_TYPE]
+                        );
+                    } catch (LocalizedException $e) {
+                        $item[CrossBorderClassInterface::CROSS_BORDER_TYPE] = CrossBorderClassInterface::NO_CROSS_BORDER_TYPE_TEXT;
```
Comment (erikhansen) [inline Ui/Component/Listing/Columns/CrossBorderClassActions.php]: Limit line length to 120 characters

File: etc/di.xml [hunk 7]
```diff
@@ -270,4 +270,9 @@
         <plugin sortOrder="1" name="add_tax_details"/>
         <plugin sortOrder="2" name="classy_llama_ava_tax_grand_total_details_sorter" type="ClassyLlama\AvaTax\Plugin\Model\Quote\GrandTotalDetailsSorter"/>
     </type>
+
```
Comment (erikhansen) [inline etc/di.xml]: Remove extra newline

File: view/frontend/layout/checkout_cart_index.xml [hunk 8]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
```
Comment (erikhansen) [inline view/frontend/layout/checkout_cart_index.xml]: Change copyright

File: view/frontend/web/template/checkout/cart/totals/grand-total.html [hunk 9]
```diff
@@ -0,0 +1,39 @@
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
```
Comment (erikhansen) [inline view/frontend/web/template/checkout/cart/totals/grand-total.html]: Change copyright

File: view/frontend/web/template/checkout/cart/totals/grand-total.html [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+<tr class="grand totals">
+    <th class="mark" scope="row">
+        <strong data-bind="i18n: title"></strong>
+    </th>
+    <td data-bind="attr: {'data-th': title}" class="amount">
+        <strong><span class="price" data-bind="text: getValue()"></span></strong>
+    </td>
+</tr>
+<!-- /ko -->
+<!-- ko foreach: totals().extension_attributes.avatax_messages -->
```
Comment (erikhansen) [inline view/frontend/web/template/checkout/cart/totals/grand-total.html]: Add "BEGIN/END EDIT" comments to make it clear what's changed

File: view/frontend/web/template/checkout/summary/grand-total.html [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+<!-- /ko -->
+<!-- ko if: isBaseGrandTotalDisplayNeeded() && isDisplayed() -->
+<tr class="totals charge">
+    <th class="mark" data-bind="i18n: basicCurrencyMessage" scope="row"></th>
+    <td class="amount">
+        <span class="price" data-bind="text: getBaseValue(), attr: {'data-th': basicCurrencyMessage}"></span>
+    </td>
+</tr>
+<!-- /ko -->
+<!-- ko foreach: totals().extension_attributes.avatax_messages -->
```
Comment (erikhansen) [inline view/frontend/web/template/checkout/summary/grand-total.html]: Add "BEGIN EDIT" comments to make it clear what's changed

File: Ui/Component/Listing/Columns/CrossBorderClassActions.php (hunk 6 above)
Comment (LordZardeck) [reply Ui/Component/Listing/Columns/CrossBorderClassActions.php]: Although this does exceed the line length limit, it is properly formatted. If this is an issue, we should discuss changes to the IDE formatter.

**Pull request merged**